python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,155 @@
+        # Reformat input
+        coordinates = f"{x}, {y}"
+        # Reproject coordinates
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent["e"], extent["n"], proj)
+        south, west = self._convert_coordinates(extent["w"], extent["s"], proj)
+        extent = {"north": north, "south": south, "east": east, "west": west}
+        return extent
+
+    def _folium_bounding_box(self, extent):
+        """Reformats extent into bounding box to pass to folium"""
+        return [[extent["north"], extent["west"]], [extent["south"], extent["east"]]]
+
+    def add_vector(self, name):
+        """Imports vector into temporary WGS84 location,
+        re-formats to a GeoJSON and adds to folium map"""
+
+        # Find full name of vector
+        file_info = gs.find_file(name, element="vector")
+        full_name = file_info["fullname"]
+        name = file_info["name"]
+        # Reproject vector into WGS84 Location
+        env_info = gs.gisenv(env=os.environ)
+        gs.run_command(
+            "v.proj",
+            input=full_name,
+            location=env_info["LOCATION_NAME"],
+            dbase=env_info["GISDBASE"],
+            env=self._vector_env,
+        )
+        # Convert to GeoJSON
+        json_file = self.tmp_dir / f"tmp_{name}.json"
+        gs.run_command(
+            "v.out.ogr",
+            input=name,
+            output=json_file,
+            format="GeoJSON",
+            env=self._vector_env,
+        )
+        # Import GeoJSON to folium and add to map
+        folium.GeoJson(json_file, name=name).add_to(self.map)
+
+    def add_layer_control(self, **kwargs):
+        self.layer_control = True
+        self.layer_control_object = folium.LayerControl(**kwargs)
+
+    def show(self):
+        """This function creates a folium map with a GRASS raster
+        overlayed on a basemap.
+
+        If map has layer control enabled, additional layers cannot be
+        added after calling show()."""
+
+        if self.layer_control == True:
```

Comment:
You need to set this to False in init to make this line work or set self.layer_control_object to None and than test it here. In both cases, the test can be just `if self.layer_control:`.